src/internal/NeoEsp32I2sXMethod.h

```diff
@@ -241,27 +240,52 @@ class NeoEsp32I2s0Mux8Bus
         //  encode bit #   0        1        2        3
         //  value zero     1        0        0        0
         //  value one      1        1        1        0    
-        const uint32_t EncodedZeroBit = 0x80000000;
-        const uint32_t EncodedOneBit = 0x80808000;
+        const uint32_t EncodedZeroBit = 0x00800000;
+        const uint32_t EncodedOneBit = 0x80800080;
         const uint32_t EncodedBitMask = 0x80808080;
 
+        const uint64_t EncodedZeroBit64 = 0x0000000000000001;
+        const uint64_t EncodedOneBit64 = 0x0000000100010001;
+        const uint64_t EncodedBitMask64 = 0x0001000100010001;
+
+        const uint64_t EncodedZeroBit64Inv = 0x0000000001000000;
+        const uint64_t EncodedOneBit64Inv = 0x0100000001000100;
+        const uint64_t EncodedBitMask64Inv = 0x0100010001000100;
+
         uint32_t* pDma = s_context.I2sBuffer;
+        uint64_t* pDma64 = reinterpret_cast<uint64_t*>(s_context.I2sBuffer);
+
         const uint8_t* pEnd = data + sizeData;
         for (const uint8_t* pPixel = data; pPixel < pEnd; pPixel++)
         {
-            uint8_t value = *pPixel++;
+            uint8_t value = *pPixel;
+
             for (uint8_t bit = 0; bit < 8; bit++)
             {
-                uint32_t dma = *(pDma);
+				if (I2sBusNumber == 0)
```

Comment:
more branching based on bus, add comment to revaulate.
These are the sorts of things that would get templatized.

Reply:
Your change is fine, The comment I request is so that I review in more depth after this merge to see if it can be cleaned up.

Reply:
OK, the comment was added